packages/react/src/ReactElement.js

```diff
@@ -164,6 +164,91 @@ const ReactElement = function(type, key, ref, self, source, owner, props) {
   return element;
 };
 
+/**
+ * https://github.com/reactjs/rfcs/pull/107
+ */
+export function jsx(type, config, maybeKey /*, source, self */) {
+  let propName;
+
+  // Reserved names are extracted
+  const props = {};
+
+  let key = null;
+  let ref = null;
+  let self = null;
+  let source = null;
+
+  // TODO: I don't allow null/undefined at dev time
+  // I think that decision makes sense because the goal is to pass props
+  // straight through. Is it React's responsibility to create an empty object
+  // if you pass null/undefined?
```

Comment:
Yea, I think we can assume that the config is always defined since this should not be called directly by users. It might be nice if empty props shared the same empty object for easier bailout but that could be guaranteed by the compiler e.g. by importing a shared object. That way we don't have to branch for it here.